src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java

```diff
@@ -680,12 +687,24 @@ private BookingTO bookingToDTOWithNullLocations(Booking booking) {
                         .updateDocumentStatusForCarrierBookingRequestReference(
                             DocumentStatus.CANC, carrierBookingRequestReference)
                         .flatMap(verifyCancellation),
-                    shipmentEventFromBooking.apply(booking)))
-        .flatMap(t -> shipmentEventService.create(t.getT2()))
-        .flatMap(shipmentEvent -> Mono.empty());
+                    Mono.just(booking)))
+        .flatMap(t -> createShipmentEventFromBooking(t.getT2()))
```

Comment:
Here `t.getT2()` is the booking without the status update to `CANC` right? So the shipment event created won't have the `CANC` status in `ShipmentEventTypeCode`. Is that intended?

Reply:
correct, fixed